include/libtorrent/kademlia/dht_tracker.hpp

```diff
@@ -128,18 +132,33 @@ namespace libtorrent { namespace dht {
 		std::vector<std::pair<node_id, udp::endpoint>> live_nodes(node_id const& nid);
 
 	private:
+		struct tracker_node
+		{
+			tracker_node(io_service& ios
+				, aux::session_listen_socket* s, socket_manager* sock
+				, libtorrent::dht_settings const& settings
+				, node_id const& nid
+				, dht_observer* observer, counters& cnt
+				, get_foreign_node_t get_foreign_node
+				, dht_storage_interface& storage);
+
+			node dht;
+			deadline_timer connection_timer;
+		};
+		using tracker_nodes_t = std::map<aux::session_listen_socket*, tracker_node>;
```

Comment:
it looks like msvc is trying to generate a copy constructor for this map, failing because ``tracker_node`` is not copyable.